src/main/scala/edu/cmu/cs/obsidian/codegen/CodeGenYul.scala

```diff
@@ -283,11 +316,13 @@ object CodeGenYul extends CodeGenerator {
                     s.flatMap(s => {
                         val id_s: Identifier = nextTemp()
                         decl_0exp(id_s) +: translateStatement(s, Some(id_s.name), contractName, checkedTable)
+                        //todo: this also does not assign afterwards; likely the same bug as fixed in IfThenElse
                     })
 
                 decl_0exp(id_scrutinee) +:
                     scrutinee_yul :+
                     edu.cmu.cs.obsidian.codegen.If(id_scrutinee, Block(s_yul))
+            // todo: also no assignment here
```

Comment:
Is there an issue for this?

Reply:
if it's wrong, it's a manifestation of the same bug as above, so i've noted it in https://github.com/mcoblenz/Obsidian/issues/339